### PR TITLE
feat: functions to refresh configstore from github

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
Next step: wire up these functions to respond to changes to github repos, and a once a day cron job.

### CODE REVIEW

File: packages/owl-bot/test/handlers.ts [hunk 1]
```diff
@@ -166,3 +181,409 @@ describe('handlers', () => {
     });
   });
 });
+
+class FakeConfigStore implements ConfigsStore {
+  readonly configs: Map<string, Configs>;
```
Comment (bcoe) [inline packages/owl-bot/test/handlers.ts:186]: I like this approach of exposing `configs` as a readonly for assertions.

**Review by bcoe — APPROVED** [review]
Non blocking nit: I found the logic that delays updates if now installationID is found a bit complex, was wondering if we could simplify to just checking both sources of installation ID.

Logistically, I think we could start with wiring up the hour schedule for syncing, identically to how we approach it here:

https://github.com/googleapis/repo-automation-bots/blob/master/packages/failurechecker/src/failurechecker.ts#L52

The scheduled task can have an installation ID associated with it -- which could be used as the default organization.

**Comment by SurferJeffAtGoogle** [comment]
> Non blocking nit: I found the logic that delays updates if now installationID is found a bit complex, was wondering if we could simplify to just checking both sources of installation ID.
> 
> Logistically, I think we could start with wiring up the hour schedule for syncing, identically to how we approach it here:
> 
> https://github.com/googleapis/repo-automation-bots/blob/master/packages/failurechecker/src/failurechecker.ts#L52
> 
> The scheduled task can have an installation ID associated with it -- which could be used as the default organization.

That would simplify the logic a lot.  I'll simplify the logic with that assumption in a follow-up CL.

File: packages/owl-bot/src/handlers.ts [hunk 2]
```diff
@@ -15,10 +15,22 @@
...
+  owlBotYamlPath,
+} from './config-files';
 import {Configs, ConfigsStore} from './configs-store';
-import {getAuthenticatedOctokit, OctokitType} from './core';
+import {getAuthenticatedOctokit, OctokitType, core} from './core';
 import {Octokit} from '@octokit/rest';
+import yaml from 'js-yaml';
+// Conflicting linters think the next line is extraneous or necessary.
+// eslint-disable-next-line node/no-extraneous-import
+import {Endpoints} from '@octokit/types';
```
[the remark this answers is not in the document]
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/src/handlers.ts:31]: But without this additional import statement, VSCode shows me a squiggly under Endpoints and gives me no intellisense.

File: packages/owl-bot/src/handlers.ts [hunk 3]
```diff
@@ -137,3 +149,178 @@ export async function createOnePullRequestForUpdatingLock(
...
+          octokit,
+          githubOrg,
+          repo.name,
+          defaultBranch,
+          installationId
+        );
+      };
+
+      // Observe the installationId.
+      if (configs?.installationId) {
```
[the remark this answers is not in the document]
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/src/handlers.ts:203]: I'll simplify the logic based on our new assumption that the task has an installationId.

File: packages/owl-bot/src/owl-bot-yaml-schema.json [hunk 4]
```diff
@@ -3,7 +3,6 @@
   "title": ".OwlBot.yaml Settings Config",
   "description": "Schema for defining the .OwlBot.yaml config",
   "additionalProperties": false,
-  "required": ["docker"],
```
[the remark this answers is not in the document]
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/src/owl-bot-yaml-schema.json:6]: Theoretically, some repo maintainers may want to copy files from googleapis-gen but never use a docker file.

**Review by SurferJeffAtGoogle — COMMENTED** [review]
Thanks for the review!  I'll merge and then simplify the logic in a follow-up PR.

**Pull request merged**